### PR TITLE
Refuse to allow insertion of bad workertype definitions

### PR DESCRIPTION
**Pull request by jhford**
@jonasfj how do you feel about this?  I should still probably put a try/catch around the requestSpotInstances and check if the failure to request a spot instance is because of a bad ami id, and if so, ignore the error and maybe send a pulse message or something?

fyi @petemoore and grenade, but I don't know his github.

### CODE REVIEW

File: lib/aws-manager.js [hunk 1]
```diff
@@ -1037,12 +1037,20 @@ class AwsManager {
...
+    try {
+      let spotRequest = await this.ec2.requestSpotInstances.inRegion(bid.region, {
+        InstanceCount: 1,
+        Type: 'one-time',
+        LaunchSpecification: launchInfo.launchSpec,
+        SpotPrice: bid.price.toString(),
+      });
+    } catch (err) {
+      if (err.code === 'InvalidAMIID.NotFound') {
+        debug(err.message + 'in ' + bid.region);
```
Comment (jonasfj) [inline lib/aws-manager.js]: probably using `console.log` I'm starting to think we should just use that for things that aren't debug messages...

Comment (jonasfj) [inline lib/aws-manager.js]: nit...

File: lib/routes/v1.js [hunk 2]
```diff
@@ -59,6 +60,97 @@ let api = new base.API({
...
+    }));
+  }));
+
+  // Finally, let's verify that the image for this workerType exists in EC2
+
+  if (reasons.length > 0) {
+    debug('Found errors with ' + workerTypeName); 
+    let e = new Error('Refusing to create an invalid worker type');
+    e.reasons = reasons;
+    throw e
```
Comment (jonasfj) [inline lib/routes/v1.js]: @jhford, this okay... But really you shouldn't communicate it this way...

`validateWorkerType` should just return a string message if there is an error, or a `null` if there is no error..
Or something like this, you generally shouldn't use exceptions to communicate return value.

Comment (jonasfj) [inline lib/routes/v1.js]: Particularly because an unhandled exception is more of a 500 and than a 400

**Comment by petemoore** [comment]
Thanks for taking care of this @jhford :+1:

**Pull request merged**